### PR TITLE
feat: [M3-7086] - Add pricing Docs Link to create/clone flows and remove DC-specific pricing warning notice

### PR DESCRIPTION
**Pull request by mjac0bs**
## Description 📝
Post-launch and after two full billing cycles, we are going to remove the info notice below the region selector that appears when a region with DC-specific pricing (currently: Jakarta; Sao Paulo) is selected and add a docs link "How Datacenter Pricing Works". The docs link was present in initial mocks, but determined to be redundant while the notice with "Learn more" existed. 

## Changes  🔄
- Removed the warning notice when a DC-specific pricing region is selected.
- Added the permanent docs link to the pricing page in the region select section of Linode/Kube/NodeBalancer Create (and Linode Clone) flows.
   -  In Kube Create, UX has signed off on placement of the docs link and moving the speedtext helper text below the region select. This looked like the cleanest placement for a form that doesn't quite match the others, since it is all in one paper and lacks a panel header.
- Cleaned up test integration and unit test coverage to test for the correct docs link, not the notice.

## Preview 📷
| Location | Before  | After   |
| ------- |------- | ------- |
| Linode Create / Distros | Screenshot 2023-12-01 at 1 08 35 PM | Screenshot 2023-12-01 at 1 09 11 PM |
| Linode Create / Clone from DC-specifc-pricing linode | Screenshot 2023-12-01 at 1 15 38 PM | Screenshot 2023-12-01 at 1 15 47 PM |
| Linode Create / Clone from non-DC-specific-pricing linode but change the region | Screenshot 2023-12-01 at 1 19 12 PM | Screenshot 2023-12-01 at 1 20 28 PM |
| NodeBalancer Create  | Screenshot 2023-12-01 at 1 01 09 PM | Screenshot 2023-12-01 at 1 00 46 PM |
| Kube Cluster Create  | Screenshot 2023-12-01 at 1 10 04 PM | Screenshot 2023-12-01 at 1 09 42 PM |

### Prerequisites
(How to setup test environment)
- Create a linode in Jakarta or Sao Paulo, if you do not have one.
- Create a linode in anywhere other than Jakarta or Sao Paolo, if you do not have one. 
- Check out this PR and `yarn up`.

### Verification steps 
(How to verify changes)
- Go to each of the create pages shown in the Preview above.
   - Confirm that you *do* see a Docs Link to the right of the region select that says `How Data Center Pricing Works` and links to https://www.linode.com/pricing.
   - Select Jakarta or Sao Paulo from the dropdown and confirm you do not see a warning notice that says `Prices for plans, products, and services in ... may vary from other regions. Learn more.`
- For the Clone flow, confirm:
   - If you are cloning from a Jakarta or Sao Paulo linode (the one you made in the Prerequisite step), you should not see the `Prices for...` notice.
   - If you are cloning from a non-DC-specific-pricing linode (e.g. located in Seattle) and you change the region to Jakarta or Sao Paulo, you *should* still see the `The selected region has a different price structure. Learn more.` notice. This is a situation where users will incur different charges for that linode if they proceed, so we want to keep this notice.
- Ensure the following tests pass in CI (and test them locally, if you'd like):

Cypress:
```
yarn cy:run -s "cypress/e2e/core/linodes/clone-linode.spec.ts,cypress/e2e/core/linodes/create-linode.spec.ts,cypress/e2e/core/nodebalancers/smoke-create-nodebal.spec.ts,cypress/e2e/core/kubernetes/lke-create.spec.ts"
```

Unit:
```
yarn test SelectRegionPanel
```

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/DynamicPriceNotice.tsx
Comment (mjac0bs) [inline packages/manager/src/components/DynamicPriceNotice.tsx:1]: This was the component for the notice. It's no longer being used, so we can say goodbye.

File: packages/manager/src/components/SelectRegionPanel/SelectRegionPanel.tsx [hunk 1]
```diff
@@ -76,22 +72,6 @@ export const SelectRegionPanel = (props: SelectRegionPanelProps) => {
...
-    !showClonePriceWarning && // Don't show both notices at the same time.
-    selectedRegionHasUniquePricing;
-
-  // If this component is used in the context of Linodes,
-  // use Linode types from the API to determine if the region
-  // has specific pricing. Otherwise, check against our local pricing map.
-  const showRegionPriceNotice = isLinode
-    ? showUniquePricingNotice
-    : selectedId && priceIncreaseMap[selectedId];
-
```
Comment (mjac0bs) [inline packages/manager/src/components/SelectRegionPanel/SelectRegionPanel.tsx:94]: All of this was to determine whether we should show the notice. 👋🏼

File: packages/manager/src/utilities/pricing/linodes.test.ts [hunk 2]
```diff
@@ -274,33 +273,3 @@ describe('getLinodeRegionPrice', () => {
     });
   });
 });
-
```
Comment (mjac0bs) [inline packages/manager/src/utilities/pricing/linodes.test.ts:277]: Removed this function, so I removed the tests.

File: packages/manager/.changeset/pr-9946-changed-1701467724778.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@linode/manager": Changed
+---
+
+Add pricing Docs Link to create/clone flows and remove DC-specific pricing warning notice  ([#9946](https://github.com/linode/manager/pull/9946))
```
Comment (mjac0bs) [inline packages/manager/.changeset/pr-9946-changed-1701467724778.md:5]: Should I separate these into the Added and Removed sections? Open to whatever we prefer.

Comment (hana-linode) [reply packages/manager/.changeset/pr-9946-changed-1701467724778.md:5]: I think it makes more sense to leave it in Changed vs two sections that point to the same PR

**Review by hana-linode — APPROVED** [review]
Confirmed that the pricing Docs link is present in
- Linode Create
- Linode Clone from same & diff region
- Nodebalancer Create
- Kube Create

Confirmed the removal of the price warning notice for all the flows above except for Linode Clone from a non-DC-specific-pricing linode to Jakarta or Sao Paulo

Unit test and e2e passed

**Review by dwiley-akamai — APPROVED** [review]
Unit & E2E tests pass ✅
Verification steps ✅
Code review ✅

**Pull request merged**